lib/cli.mjs

```diff
@@ -526,13 +561,64 @@ export default async function main(argv) {
 		options.computeEditInfo = computeEditInfo;
 	}
 
+	if (isBoolean(suppressPrune)) {
+		options.suppressPrune = suppressPrune;
+	}
+
+	// Handle suppress logic - detect both --suppress (all) and --suppress=rule cases
+	const hasPlainSuppress = argv.includes('--suppress');
+	const hasRuleSuppress = Array.isArray(suppress) && suppress.length > 0;
+
+	if (hasPlainSuppress) {
+		options.suppressAll = true;
+	}
+
+	if (hasRuleSuppress) {
+		options.suppressRule = suppress;
+	}
+
+	if (isString(suppressLocation)) {
+		options.suppressLocation = suppressLocation;
+	}
+
+	if (options.suppressAll && options.suppressRule) {
+		process.stderr.write(
+			`${red('Error: The --suppress and --suppress=<rule> options cannot be used together.')}${EOL}`,
+		);
+		process.exitCode = EXIT_CODE_INVALID_USAGE;
+
+		return;
+	}
+
+	if ((options.suppressAll || options.suppressRule) && options.suppressPrune) {
+		process.stderr.write(
+			`${red('Error: The --suppress and --suppress-prune options cannot be used together.')}${EOL}`,
+		);
+		process.exitCode = EXIT_CODE_INVALID_USAGE;
+
+		return;
+	}
+
 	// Add input/code into options
 	if (cli.input.length > 0) {
 		options.files = cli.input;
 	} else {
 		options.code = await getStdin();
 	}
 
+	const hasStdin = stdin || isString(options.code);
+
+	if (hasStdin && (options.suppressAll || options.suppressRule || options.suppressPrune)) {
+		process.stderr.write(
+			`${red(
+				'Error: The --suppress, --suppress=<rule>, and --suppress-prune options cannot be used with piped-in code.',
```

Comment:
question

Does pruning has anything to do with the current target of the linting?
If not, it should be allowed even with piped-in code, correct?

Reply:
Could you provide a specific script example for that use case? 

I'm not sure why we need to support pruning with stdin when a simple approach like this works perfectly:

```json
{ 
   "scripts": {
       "lint:prune": "stylelint --suppress-prune '**/*.css'"
    }
}
```